commons/ava_testsuite/five_node_staking_gecko_network_tests.go

```diff
@@ -0,0 +1,73 @@
+package ava_testsuite
+
+import (
+	"github.com/kurtosis-tech/ava-e2e-tests/commons/ava_networks"
+	"github.com/kurtosis-tech/ava-e2e-tests/gecko_client"
+	"github.com/kurtosis-tech/kurtosis/commons/testsuite"
+	"github.com/palantir/stacktrace"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+
+type FiveNodeStakingNetworkBasicTest struct{}
+func (test FiveNodeStakingNetworkBasicTest) Run(network interface{}, context testsuite.TestContext) {
+	castedNetwork := network.(ava_networks.NNodeGeckoNetwork)
+
+	// TODO check ALL nodes!
+	client, err := castedNetwork.GetGeckoClient(0)
+	if err != nil {
+	context.Fatal(stacktrace.Propagate(err, "Could not get client"))
+	}
+
+	peers, err := client.AdminApi().GetPeers()
+	if err != nil {
+	context.Fatal(stacktrace.Propagate(err, "Could not get peers"))
+	}
+
+	context.AssertTrue(len(peers) == 9)
+	}
+
+	func (s FiveNodeStakingNetworkBasicTest) GetNetworkLoader() (testsuite.TestNetworkLoader, error) {
+		return ava_networks.NewNNodeGeckoNetworkLoader(5, 3, true)
+	}
+
+
+	// =============== Get Validators Test ==================================
+	type FiveNodeStakingNetworkGetValidatorsTest struct{}
+	func (test FiveNodeStakingNetworkGetValidatorsTest) Run(network interface{}, context testsuite.TestContext) {
+		castedNetwork := network.(ava_networks.NNodeGeckoNetwork)
+
+		// TODO we need to make sure ALL the nodes agree about validators!
+		client, err := castedNetwork.GetGeckoClient(0)
+		if err != nil {
+			context.Fatal(stacktrace.Propagate(err, "Could not get client"))
+		}
+
+		// TODO This retry logic is only necessary because there's not a way for Ava nodes to reliably report
+		//  bootstrapping as complete; remove it when Gecko can report successful bootstrapping
+		var validators []gecko_client.Validator
+		for i := 0; i < 5; i++ {
+			validators, err = client.PChainApi().GetCurrentValidators()
+			if err == nil {
+				break
+			}
+			logrus.Error(stacktrace.Propagate(err, "Could not get current validators; sleeping for 5 seconds..."))
+			time.Sleep(5 * time.Second)
+		}
+		// TODO This should go away as soon as Ava can reliably report bootstrapping as complete
+		if validators == nil {
+			context.Fatal(stacktrace.NewError("Could not get validators even after retrying!"))
+		}
+
+		for _, validator := range validators {
+			logrus.Infof("Validator ID: %s", validator.Id)
+		}
+		// TODO change this to be specific
+		context.AssertTrue(len(validators) >= 1)
+}
+
+func (test FiveNodeStakingNetworkGetValidatorsTest) GetNetworkLoader() (testsuite.TestNetworkLoader, error) {
+	return ava_networks.NewNNodeGeckoNetworkLoader(5, 3, true)
```

Comment:
This should break - it has 3 boot nodes, and we only have one cert hardcoded! This would mean that we now have multiple nodes with the same ID!

Reply:
pushed change, done. they wont get the same id actually, we just wont have ids for the other two and they wont be used as bootstrapper technically